### PR TITLE
Documentation tests example

### PR DESCRIPTION
**Pull request by apiraino**
just discovered this

### CODE REVIEW

File: rust/language-reference/7-tests.md [hunk 1]
```diff
@@ -135,5 +136,37 @@ Since integration tests sit in a completely different crate than our project, th
...
+## Documentation tests
+
+These tests can be a quick way to embed documentation *and* unit tests at the same time.
+These tests have a special `//!` comment placed at the module level and can be run with `cargo tests`.
+You can then run `cargo doc --no-deps` and build the documentation too.
+
+```
+//! This is a documentation test
+//!
+//! ```                                                               .
```
Comment (michelezamuner) [inline rust/language-reference/7-tests.md]: The tags to open codeblocks seem wrong, also below.

File: rust/language-reference/7-tests.md [hunk 2]
```diff
@@ -135,5 +136,37 @@ Since integration tests sit in a completely different crate than our project, th
...
+//!
+//! # Example of how my module works
+//!
+//! let a = 5;
+//! let b = 6;
+//!
+//! assert_eq!(11, sum(a + b));
+//! ```                                                               .
+
+fn sum(a: i32, b: i32) -> i32 {
```
Comment (michelezamuner) [inline rust/language-reference/7-tests.md]: Should these comments go in the test file, or in the source file? Here it looks like they're put in the source.

File: rust/language-reference/7-tests.md [hunk 3]
```diff
@@ -135,5 +136,37 @@ Since integration tests sit in a completely different crate than our project, th
 Also, it's important to notice that it's necessary for the project to have a `lib.rs` file (i.e. to be exposed as a library) for integration tests to work, because otherwise it would be impossible to import the project as an external crate inside the `tests` crate.
 
 
+## Documentation tests
+
+These tests can be a quick way to embed documentation *and* unit tests at the same time.
+These tests have a special `//!` comment placed at the module level and can be run with `cargo tests`.
```
Comment (michelezamuner) [inline rust/language-reference/7-tests.md]: What happens when we run `cargo test` (I believe there's a typo too) with these tests in place?

File: rust/language-reference/7-tests.md (hunk 2 above)
Comment (apiraino) [reply rust/language-reference/7-tests.md]: Correct, they are in any source file, that's how the magic happens. I wrote a [complete crate example here](https://github.com/apiraino/rust-module-doc-tests).

Styling markdown here is not easy, there are overlapping backticks and other typos; I'll try to fix it.

File: rust/language-reference/7-tests.md (hunk 3 above)
Comment (apiraino) [reply rust/language-reference/7-tests.md]: acknowledge the typo, sorry. These "documentation" tests will be run together along your tests marked by the usual `#cfg[test]` tag.

**Pull request merged**